src/gpodder/gtkui/main.py

```diff
@@ -1963,125 +1978,57 @@ def treeview_available_show_context_menu(self, treeview, event=None):
+            self.episode_context_menu_helper.replace_entries(entries)
 
-            menu.append(Gtk.SeparatorMenuItem())
-
-            item = Gtk.CheckMenuItem(_('New'))
-            item.set_active(any_new)
-            if any_new:
-                item.connect('activate', lambda w: self.mark_selected_episodes_old())
+            # 'Send to' submenu
+            if downloaded:
+                if self.sendto_menu.get_n_items() < 1:
+                    self.sendto_menu.insert_submenu(
+                        0, _('Send to'),
+                        self.application.builder.get_object('episodes-context-sendto'))
             else:
-                item.connect('activate', lambda w: self.mark_selected_episodes_new())
-            menu.append(item)
-
-            if can_lock:
-                item = Gtk.CheckMenuItem(_('Archive'))
-                item.set_active(any_locked)
-                item.connect('activate',
-                             lambda w: self.on_item_toggle_lock_activate(
-                                 w, False, not any_locked))
-                menu.append(item)
+                self.sendto_menu.remove_all()
 
-            menu.append(Gtk.SeparatorMenuItem())
-            # Single item, add episode information menu item
-            item = Gtk.ImageMenuItem(_('Episode details'))
-            item.set_image(Gtk.Image.new_from_icon_name('dialog-information',
-                                                        Gtk.IconSize.MENU))
-            item.set_action_name('win.toggleShownotes')
-            menu.append(item)
-
-            if len(self.get_selected_episodes()) == 1:
-                item = Gtk.MenuItem(_('Open download folder'))
-                item.connect('activate', self.on_open_episode_download_folder)
-                menu.append(item)
+            # New and Archive state
+            self.gPodder.lookup_action('episodeNew').change_state(GLib.Variant.new_boolean(any_new))
+            self.gPodder.lookup_action('episodeLock').change_state(GLib.Variant.new_boolean(any_locked))
 
-            menu.attach_to_widget(treeview)
-            menu.show_all()
-            # Disable tooltips while we are showing the menu, so
-            # the tooltip will not appear over the menu
-            self.treeview_allow_tooltips(self.treeAvailable, False)
-            menu.connect('deactivate', lambda menushell: self.treeview_allow_tooltips(self.treeAvailable, True))
             if event is None:
-                func = TreeViewHelper.make_popup_position_func(treeview)
-                menu.popup(None, None, func, None, 3, Gtk.get_current_event_time())
+                area = TreeViewHelper.get_popup_rectangle(treeview)
             else:
-                menu.popup(None, None, None, None, event.button, event.time)
+                area = Gdk.Rectangle()
+                area.x, area.y = event.x, (event.y + 25)  # adjust for treeView header
```

Comment:
Is 25 the height of header on your system? Fixed values like that fail when users have different font sizes or themes.

Reply:
Also, when opened from menu key, the header height isn't accounted for, causing the menu to be higher up than it should.

Reply:
Thanks for catching this, I forgot to fix this hack. Now it should be correct.